transport/tchannel/config.go

```diff
@@ -145,6 +157,10 @@ func (ts *transportSpec) buildInbound(c *InboundConfig, t transport.Transport, k
 	}
 
 	trans.addr = c.Address
+	// Override inbound TLS mode when not set by an option.
+	if trans.inboundTLSMode == nil {
+		trans.inboundTLSMode = &c.TLS.Mode
```

Comment:
I did not see a nil check on `c` here before we `&c`. Should we do that?

Reply:
`c` is accessed without nil check throughout `buildInbound` method, reflect seems to be always providing a non nil parameter

Reply:
interesting. I'm fine the way it is for now because the previous line is just accessing `c`. But we should look into it and make sure we don't have nil pointer in the code path.